parsers/ES.py

```diff
@@ -113,17 +158,50 @@ def check_valid_parameters(
 
 
 def fetch_island_data(
-    zone_key: ZoneKey, session: Session, target_datetime: Optional[datetime]
-) -> List[Response]:
+    zone_key: ZoneKey,
+    session: Session,
+    target_datetime: Optional[datetime],
+    fuel_mapping: Dict = FUEL_MAPPING,
+):
     """Fetch data for the given zone key."""
+    timezone = TIMEZONES_MAPPING[ZONE_FUNCTION_MAP[zone_key]]
     if target_datetime is None:
-        date = target_datetime
+        date = utcnow().to(timezone).format("YYYY-MM-DD")
     else:
         date = target_datetime.strftime("%Y-%m-%d")
-
-    data: List[Response] = ZONE_FUNCTION_MAP[zone_key](session).get_all(date)
-    if data:
-        return data
+    system = ZONE_MAPPING[ZONE_FUNCTION_MAP[zone_key]]
+    zone = API_CODE_MAPPING[ZONE_FUNCTION_MAP[zone_key]]
+    res = session.get(
+        f"https://demanda.ree.es/WSvisionaMoviles{system}Rest/resources/demandaGeneracion{system}?curva={zone}&fecha={date}"
+    )
+    if not res.ok:
+        raise ParserException(
+            "ES.py",
+            f"Failed fetching data for {zone_key}",
+            zone_key,
+        )
+    json = loads(res.text.replace("null(", "", 1).replace(r");", "", 1))
+    data = json["valoresHorariosGeneracion"]
+    responses = []
+    for value in data:
+        ts = value.pop("ts")
+        arrow = get(ts + " " + timezone, "YYYY-MM-DD HH:mm ZZZ")
+        response = {}
+        response["timestamp"] = str(arrow)
+        for key in value:
+            if key in fuel_mapping:
+                if fuel_mapping[key] in response.keys():
+                    response[fuel_mapping[key]] += value[key]
+                else:
+                    response[fuel_mapping[key]] = value[key]
+            elif key in LINK_MAPPING:
+                if LINK_MAPPING[key] in response.keys():
+                    response[LINK_MAPPING[key]] += value[key]
+                else:
+                    response[LINK_MAPPING[key]] = value[key]
```

Comment:
This should not be aggregated here. We should aggregate it later on using the parser classes, they contain built in validation to ensure data quality which will be degraded if we aggregate it here first.